scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -528,6 +515,9 @@ def mark_to_warm(self, container_id, pod_namespace, pod_name):
         try:
             if not container_id in self.__active_pods:
                 self.__active_pods[container_id] = ControlledCacheWarmer.WarmingEntry(pod_namespace, pod_name)
+            else:
+                warming_entry = self.__active_pods[container_id]
+                warming_entry.is_warm = self.is_warm(pod_namespace, pod_name)
```

Comment:
Had to add this to propagate cache invalidations into the warming entries.  Otherwise, warming entries seem to be always True and come cache interval time, the `k8s_cache.invalidate` does not result in any re-querying of API master.